_contrib/travis-check-png-optimized.sh

```diff
@@ -0,0 +1,17 @@
+#!/bin/sh
+
+## Finds newly added or modified PNG files in the repository and checks if each file is optimized.
+## If not optimized, it fails with an error code of 1
+
+CHANGEDFILES=$(git diff --name-only --diff-filter=AM "$TRAVIS_COMMIT_RANGE")
+
+for FILE in $CHANGEDFILES; do
+	if echo "$FILE" | grep -q ".png$"; then 
+		echo checking "$FILE"
+		OPTIPNGOUTPUT=$(optipng -simulate "$FILE" 2>&1)
+		if ! echo "$OPTIPNGOUTPUT" | grep -q "is already optimized"; then
+			echo "The file $FILE is not optimized. To optimize use 'optipng -o7 $FILE'."
+			exit 1
```

Comment:
I'd prefer the test identify and print a listing of all failed files instead of just the first failure, that way they can all be fixed in one go (especially useful since this could be a slow test if there are lots of files being added).  E.g. `for FILE in $CHANGEDFILES; do [...] ; done | grep -q 'is not optimized' ; if [ $? -eq 0 ]; then exit 1 ; fi`